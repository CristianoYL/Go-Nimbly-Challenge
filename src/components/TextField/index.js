import TextField from './text_field';

export default TextField;
