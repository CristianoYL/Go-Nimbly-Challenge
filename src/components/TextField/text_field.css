.text-field {
  margin: 0 0;
  display: flex;
  flex-flow: column nowrap;
}

input {
  margin: 0 auto;
  width: 100%;
  font-size: 150%;
  border: none;
  border-bottom: 1px solid grey;
  background-color: transparent;
  outline: none;
  color: var(--color-primary);
}

input:focus {
  border-bottom: 1px solid var(--color-primary);
  color: var(--color-primary);
}

.text-field label:focus-within{
  color: var(--color-primary);
}

input:focus::placeholder {
  color: transparent;
}

.text-field label {
  color: transparent;
  margin-top: 10px;
  margin-bottom: 1px;
  font-size: 70%;
}
