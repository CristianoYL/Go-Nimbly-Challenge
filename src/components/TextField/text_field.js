import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './text_field.css';

export default class TextField extends Component {
  constructor(props) {
    super(props);

    this.state = { value: '' };
  }

  onTextChange(value) {
    // define legal chars and filter
    const cleanValue = value.replace(/[^a-z0-9A-Z+\-/*().\^]/g, '');
    this.setState({ value: cleanValue });
    this.props.onTextChange(cleanValue);
  }

  render() {
    // show a hint first, then store user input into state.value
    return (
      <div className={`text-field ${this.props.className}`}>
        <label htmlFor={this.props.hint} target="_input">
          {this.props.hint}
          <input
            id={this.props.hint}
            value={this.state.value}
            placeholder={this.props.hint}
            onChange={event => this.onTextChange(event.target.value)}
          />
        </label>
      </div>
    );
  }
}

TextField.defaultProps = { className: '' };
TextField.propTypes = {
  hint: PropTypes.string.isRequired,
  onTextChange: PropTypes.func.isRequired,
  className: PropTypes.string,
};
