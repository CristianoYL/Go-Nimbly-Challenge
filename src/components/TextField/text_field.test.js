import React from 'react';
import renderer from 'react-test-renderer';
import TextField from './text_field';

test('TextField without className', () => {
  const onTextChange = () => {};
  const component = renderer.create(
    <TextField
      onTextChange={onTextChange}
      hint="test"
      />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('TextPanel with all props', () => {
  const onTextChange = () => {};
  const component = renderer.create(
    <TextField
      hint="test hint"
      onTextChange={onTextChange}
      className="test-class"
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('props.onTextChange gets called', () => {
  let oldValue = false;
  const newValue = true;
  const onTextChange = (value) => {
    oldValue = value;
  };
  const component = renderer.create(
    <TextField
      hint="test hint"
      onTextChange={onTextChange}
      className="test-class"
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component.getInstance().onTextChange(newValue);
  expect(oldValue).toEqual(newValue);
});
