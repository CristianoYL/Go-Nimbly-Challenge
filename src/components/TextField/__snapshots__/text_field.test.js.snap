// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField without className 1`] = `
<div
  className="text-field "
>
  <label
    htmlFor="test"
    target="_input"
  >
    test
    <input
      id="test"
      onChange={[Function]}
      placeholder="test"
      value=""
    />
  </label>
</div>
`;

exports[`TextPanel with all props 1`] = `
<div
  className="text-field test-class"
>
  <label
    htmlFor="test hint"
    target="_input"
  >
    test hint
    <input
      id="test hint"
      onChange={[Function]}
      placeholder="test hint"
      value=""
    />
  </label>
</div>
`;

exports[`props.onTextChange gets called 1`] = `
<div
  className="text-field test-class"
>
  <label
    htmlFor="test hint"
    target="_input"
  >
    test hint
    <input
      id="test hint"
      onChange={[Function]}
      placeholder="test hint"
      value=""
    />
  </label>
</div>
`;
